baton-maven-plugin/src/main/java/org/technologybrewery/baton/util/FileUtils.java

```diff
@@ -0,0 +1,213 @@
+package org.technologybrewery.baton.util;
+
+import org.apache.commons.io.IOUtils;
+import org.codehaus.plexus.util.StringUtils;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Common file-related logic used when performing a migration
+ */
+public final class FileUtils {
+    private static final String SPACE = " ";
+
+    private FileUtils(){
+    }
+
+    /**
+     * Evaluates a file against a literal string and replaces each match with a specified replacement
```

Comment:
A: very minor, but a period at the end of the first sentence in Javadoc really improves functionality (whole sentence shows up in the summary instead the first ~40 characters)

Reply:
The more you know!